pkg/controller/container-runtime-config/container_runtime_config_bootstrap.go

```diff
@@ -0,0 +1,96 @@
+package containerruntimeconfig
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	mcfgv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// RunContainerRuntimeBootstrap generates ignition configs at bootstrap
+func RunContainerRuntimeBootstrap(templateDir string, crconfigs []*mcfgv1.ContainerRuntimeConfig, controllerConfig *mcfgv1.ControllerConfig, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	var res []*mcfgv1.MachineConfig
+	managedKeyExist := make(map[string]bool)
+	for _, cfg := range crconfigs {
```

Comment:
Should probably validateUserContainerRuntimeConfig() sometime before or during this loop

Reply:
Fixed